ckanext/s3filestore/uploader.py

```diff
@@ -64,9 +65,20 @@ def get_directory(self, id, storage_path):
         return directory
 
     def get_s3_session(self):
-        return boto3.session.Session(aws_access_key_id=self.p_key,
+        session = boto3.session.Session(aws_access_key_id=self.p_key,
                                      aws_secret_access_key=self.s_key,
                                      region_name=self.region)
+        if self.role:
+            assumed_role_object = session.client('sts').assume_role(
+                RoleArn=self.role,
+                RoleSessionName="CkanExtS3Session")
+            credentials = assumed_role_object['Credentials']
+            return boto3.session.Session(
```

Comment:
Perhaps assign this to `session` (which is about to be returned) instead of returning it directly?

Or change line 71 into a guard clause, "if no role specified then return immediately".

Reply:
I think you've missed the part where the initial session is used to communicate with STS if there is a role configured. So in this case both sessions are needed but only the one that has the assumed role is returned to be used for communicating with S3.

Reply:
No, I didn't miss that. I'm just suggesting that it's more readable to assign and then return, instead of having two separate return statements.

(Having only one exit from a function is more of a guideline than a hard rule, but it's not a bad guideline when it's convenient.)